### PR TITLE
Minor cleanup of how multiple swapchains are managed within WindowContext

### PR DESCRIPTION
**Pull request by moudgils**
## What does this PR do?

Cleanup of how swapchain data is grouped and how it is managed in a more cohesive manner. 

## How was this PR tested?

Ran ASV full test suite and the Editor.

### CODE REVIEW

File: Gems/Atom/RPI/Code/Source/RPI.Public/WindowContext.cpp [hunk 1]
```diff
@@ -172,15 +182,27 @@ namespace AZ
...
+                m_swapChainsData.insert(
+                    m_swapChainsData.begin() + defaultSwapChainIndex, SwapChainData{ swapChain, RHI::Viewport{}, RHI::Scissor{} });
+            }
 
             // Add XR pipelines if it is active
             XRRenderingInterface* xrSystem = RPISystemInterface::Get()->GetXRSystem();
             if (xrSystem)
             {
-                for (AZ::u32 i = 0; i < xrSystem->GetNumViews(); i++)
+                AZ::u32 numXrViews = xrSystem->GetNumViews();
```
Comment (moraaar) [inline Gems/Atom/RPI/Code/Source/RPI.Public/WindowContext.cpp:203]: `numXrViews` can be const.

File: Gems/Atom/RPI/Code/Source/RPI.Public/WindowContext.cpp [hunk 2]
```diff
@@ -68,27 +71,33 @@ namespace AZ
 
         const RHI::Ptr<RHI::SwapChain>& WindowContext::GetSwapChain(SwapChainMode swapChainMode) const
         {
-            return m_swapChains[static_cast<uint32_t>(swapChainMode)];
+            uint32_t swapChainIndex = static_cast<uint32_t>(swapChainMode);
+            AZ_Assert(swapChainIndex < GetSwapChainsSize(), "Swapchain does not exist");
```
Comment (moraaar) [inline Gems/Atom/RPI/Code/Source/RPI.Public/WindowContext.cpp:75]: It'd be informative to print the index and size in the assert message. Same in the other asserts.

**Review by moraaar — APPROVED** [review]
LGTM. Some minor feedback.

File: Gems/Atom/RPI/Code/Source/RPI.Public/WindowContext.cpp [hunk 3]
```diff
@@ -68,27 +71,33 @@ namespace AZ
...
         uint32_t WindowContext::GetSwapChainsSize() const
         {
-            return aznumeric_cast<uint32_t>(m_swapChains.size());
+            return aznumeric_cast<uint32_t>(m_swapChainsData.size());
         }
 
         const RHI::Viewport& WindowContext::GetViewport(SwapChainMode swapChainMode) const
         {
-            return m_defaultViewports[static_cast<uint32_t>(swapChainMode)];     
+            uint32_t swapChainIndex = static_cast<uint32_t>(swapChainMode);
```
Comment (jeremyong-az) [inline Gems/Atom/RPI/Code/Source/RPI.Public/WindowContext.cpp:86]: Nit: Could do a slight consolidation with `return GetSwapchain(swapChainMode).m_viewport` here and below for the scissor if you want to not duplicate the assertion logic

**Review by jeremyong-az — APPROVED** [review]

**Review by antonmic — APPROVED** [review]

**Pull request merged**